### PR TITLE
add getTokenLargestAccounts rpc method to rust client

### PR DESCRIPTION
**Pull request by anthontaylor**
#### Problem
getTokenLargestAccounts is an available rpc method on the web3.js rpc client but not the rust rpc client.
https://solana-labs.github.io/solana-web3.js/classes/Connection.html#getTokenLargestAccounts

#### Summary of Changes
I've added the functionality to both the nonblocking and blocking rpc clients.

Fixes #
 #23457

### CODE REVIEW

File: client/src/nonblocking/rpc_client.rs [hunk 1]
```diff
@@ -5016,6 +5016,21 @@ impl RpcClient {
         .await
     }
 
+    pub async fn get_token_largest_accounts(
```
Comment (joncinque) [inline client/src/nonblocking/rpc_client.rs:5019]: Let's follow the model of the other `get_token_*` calls, and split it up to `get_token_largest_accounts` and `get_token_largest_accounts_with_commitment`

File: client/src/rpc_client.rs [hunk 2]
```diff
@@ -3956,6 +3956,17 @@ impl RpcClient {
         ))
     }
 
+    pub fn get_token_largest_accounts(
```
Comment (joncinque) [inline client/src/rpc_client.rs:3959]: Same here

**Review by joncinque — COMMENTED** [review]
Sorry for the late review -- this is on the right track! Just a bit more and then we can merge it

File: client/src/nonblocking/rpc_client.rs (hunk 1 above)
Comment (anthontaylor) [reply client/src/nonblocking/rpc_client.rs:5019]: Thanks for the suggestion. I went ahead and split them up.

File: client/src/rpc_client.rs (hunk 2 above)
Comment (anthontaylor) [reply client/src/rpc_client.rs:3904]: I split this one up as well

**Comment by joncinque** [comment]
This is looking ready, next we need it to pass CI. From the root of the repo, you'll need to run `./cargo nightly fmt --all` and maybe also `./cargo nightly clippy`

**Pull request merged**